ticdc/ticdc-canal-json.md

```diff
@@ -258,6 +258,23 @@ The following table shows the mapping relationships between Java SQL Types in Ti
 
 For more information about Java SQL Types, see [Java SQL Class Types](https://docs.oracle.com/javase/8/docs/api/java/sql/Types.html).
 
+## Binary and Blob types
```

Comment:
could you also describe the method of converting a byte array to a string in the documentation？

Reply:
Are you asking for a worked example in the documentation in a specific programming language like Java or Go? A lot of this would be language specific so not sure what would be useful to further document here.

Here's a minimal go program I wrote to outline the encoding steps taken (encoding the bytes with ISO8859-1, converting to a string, marshaling to JSON) without having to have a full implementation running. Not sure if part of this would address what you are asking for?

```
package main

import (
	"encoding/json"
	"fmt"

	"golang.org/x/text/encoding"
	"golang.org/x/text/encoding/charmap"
)

type Row struct {
	C1 string `json:"c1"`
}

func main() {

	value := []byte{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142, 143, 144, 145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 155, 156, 157, 158, 159, 160, 161, 162, 163, 164, 165, 166, 167, 168, 169, 170, 171, 172, 173, 174, 175, 176, 177, 178, 179, 180, 181, 182, 183, 184, 185, 186, 187, 188, 189, 190, 191, 192, 193, 194, 195, 196, 197, 198, 199, 200, 201, 202, 203, 204, 205, 206, 207, 208, 209, 210, 211, 212, 213, 214, 215, 216, 217, 218, 219, 220, 221, 222, 223, 224, 225, 226, 227, 228, 229, 230, 231, 232, 233, 234, 235, 236, 237, 238, 239, 240, 241, 242, 243, 244, 245, 246, 247, 248, 249, 250, 251, 252, 253, 254, 255}

	j1 := encodeIsoJson(value)
	b1 := decodeIsoJson(string(j1))

	fmt.Println("Encoded format")
	fmt.Println(string(j1))
	fmt.Println()
	fmt.Println("Roundtripped bytes")
	fmt.Println(b1)
	fmt.Println()
}

func encodeIsoJson(binarydata []byte) []byte {
	var decoder *encoding.Decoder = charmap.ISO8859_1.NewDecoder()
	decodedBytes, _ := decoder.Bytes(binarydata)
	stringifiedBytes := string(decodedBytes)
	dataRow := &Row{C1: stringifiedBytes}
	jsonDataRow, _ := json.Marshal(dataRow)
	return jsonDataRow
}

func decodeIsoJson(jsonString string) []byte {
	var encoder *encoding.Encoder = charmap.ISO8859_1.NewEncoder()

	jsonData := new(Row)
	json.Unmarshal([]byte(jsonString), jsonData)
	encodedBytes, _ := encoder.Bytes([]byte(jsonData.C1))
	return encodedBytes
}
```

Reply:
It would be great if there is an example, because customers have given feedback that they do not understand why the canal-json encoding for binary types results in significant data inflation.

Reply:
Okay, I have added a brief example of the impact of the encoding, including showing how the escaping mechanism adds additional characters. See https://github.com/pingcap/docs/pull/13832/commits/a81c88ee303b94dc375db47ed7167bdcfb4b510b